src/parser/rogue/subtlety/modules/features/VanishFindWeakness.tsx

```diff
@@ -0,0 +1,152 @@
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CHEAP_SHOT), this.onFindWeaknessApplierCast);
+  }
+
+  onVanishCast(event: CastEvent): void {
+      this.lastVanishPtr = event;
+  }
+
+  onFindWeaknessApplierCast(event: CastEvent): void {
+    if (event.targetID === undefined) {
+      this.debug(`Skipping FW buff tracking for FW applier cast at timestamp ${event.timestamp} due to missing target ID.`);
+      return;
+    }
+    if (this.lastVanishPtr === null) {
+      return;
+    }
+    const enemyID: number = event.targetID as number;
+    const enemy: Enemy = this.enemies.getEntities()[enemyID];
+    if (enemy === undefined) {
+      // In this case, a FW applier was cast on a non-enemy target.
+      // Maybe the rogue was mind-controlled and cast it on another player?
+      return;
+    }
+    const enemyHasFW: boolean = enemy.hasBuff(SPELLS.FIND_WEAKNESS.id, event.timestamp, 0, 0, this.selectedCombatant.id);
+    if (!enemyHasFW) {
+      // This is the first time we cast a FW applier on the target, so we can safely assume it has not had FW before.
+      // Or we've hit the target with a FW applier before, but they don't have it now.
+      return;
+    } else {
+      const fwTimeRemaining: number = enemy.getRemainingBuffTimeAtTimestamp(SPELLS.FIND_WEAKNESS.id, this.FW_DEBUFF_LENGTH, this.DEBUFF_CAP, event.timestamp, this.selectedCombatant.id);
+      if (fwTimeRemaining > this.BAD_CAST_WINDOW) {
+        this.badVanishCasts.set(this.lastVanishPtr, fwTimeRemaining);
+      }
+    }
+  }
+
+  get badVanishCastsSuggestionThresholds(): NumberThreshold {
+    return {
+      actual: this.badVanishCasts.size,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 0
+      },
+      style: ThresholdStyle.NUMBER,
+    }
+  }
+
+  suggestions(when: When) {
+    when(this.badVanishCastsSuggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => suggest(t({
+        id: "rogue.subtlety.suggestions.findWeaknessAndVanish.badCasts.suggestion",
+        message: `Avoid casting Vanish with more than ${this.BAD_CAST_WINDOW/1000}s left on Find Weakness on the current target.`
+      }))
+        .icon(SPELLS.VANISH.icon)
+        .actual(t({
+          id: "rogue.subtlety.suggestions.findWeaknessAndVanish.badCasts.actual",
+          message: `You cast Vanish ${this.badVanishCasts.size} times with more than ${this.BAD_CAST_WINDOW/1000}s left on Find Weakness on the current target.`
+        }))
+        .recommended(t({
+          id: "rogue.subtlety.suggestions.findWeaknessAndVanish.badCasts.recommended",
+          message: `Do not cast Vanish with more than ${this.BAD_CAST_WINDOW/1000}s left on Find Weakness on the current target.`
+        })));
+  }
+
+  statistic(): React.ReactNode {
+    const tableEntries: React.ReactNode[] = [];
+    const keys: CastEvent[] = Array.from(this.badVanishCasts.keys());
+    for (let i: number = 0; i < keys.length; i++) {
```

Comment:
This goes against the style guide

Reply:
Replace with `keys.forEach()` I'm guessing?